### PR TITLE
Saving settings to local storage

### LINKED ISSUE #1209: Task settings revert to default
After, for example, enabling autosave and setting the framerate for playback, the settings go back to the default values after the task is closed and reopened. Is this the expected behaviour?

### PR DESCRIPTION
**Pull request by ActiveChooN**
### Motivation and context

This PR provide saving user settings to local storage. Resolved #1209

### How has this been tested?

Manually

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
~~- [ ] I have added tests to cover my changes~~
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [Market-1501](https://www.aitribune.com/dataset/2018051063) format support (<https://github.com/openvinotoolkit/cvat/pull/2869>)
 - Ability of upload manifest for dataset with images (<https://github.com/openvinotoolkit/cvat/pull/2763>)
 - Annotations filters UI using react-awesome-query-builder (https://github.com/openvinotoolkit/cvat/issues/1418)
+- Settings saving in browser (<https://github.com/openvinotoolkit/cvat/pull/3017>)
```
Comment (bsekachev) [inline CHANGELOG.md:33]: For clarify would rephrase it to:

Storing settings in local storage to keep them between browser sessions

File: cvat-ui/src/components/header/settings-modal/settings-modal.tsx [hunk 2]
```diff
@@ -21,6 +26,44 @@ interface SettingsModalProps {
...
+                    if (!(sectionKey in newSettings)) newSettings[sectionKey] = {};
+                    newSettings[sectionKey][key] = settedValue;
+                }
+            }
+            dispatch(setSettings(newSettings));
+        } catch {
+            notification.error({
+                message: 'Failed to load settings from local storage',
+            });
+        }
```
Comment (bsekachev) [inline cvat-ui/src/components/header/settings-modal/settings-modal.tsx:64]: Why not to do it in defaultValue for settings reducer?

File: cvat-ui/src/components/header/settings-modal/settings-modal.tsx [hunk 3]
```diff
@@ -29,9 +72,16 @@ const SettingsModal = (props: SettingsModalProps): JSX.Element => {
...
-                </Button>
+                <>
+                    <Tooltip title='Will save settings from this page and appearance settings on standard workspace page in browser'>
+                        <Button type='primary' onClick={onSaveSettings}>
+                            Save
+                        </Button>
+                    </Tooltip>
+                    <Button type='default' onClick={onClose}>
+                        Close
+                    </Button>
```
Comment (bsekachev) [inline cvat-ui/src/components/header/settings-modal/settings-modal.tsx:83]: Need to update dock. Do we need button close? We have "X" and also click everywhere out of  the modal closes the modal

Comment (ActiveChooN) [reply cvat-ui/src/components/header/settings-modal/settings-modal.tsx:83]: It's default behaviour for the modal element.

**Comment by ActiveChooN** [comment]
@TOsmanov, hi, can you adjust documentation according this new feature, please?
